Dockerfile.dev

```diff
@@ -0,0 +1,11 @@
+FROM node:alpine
+
+WORKDIR /app
+
+COPY package.json /app
+
+RUN npm install
+
+COPY . /app
+
+CMD ["npm", "run", "dev"]
```

Comment:
What is the difference between Dockerfile and Dockerfile.dev

Reply:
> What is the difference between Dockerfile and Dockerfile.dev

`Dockerfile` is for developing the production instance of the application. `Dockerfile.dev` is for rapid prototyping with the dev instance of the application. If you check the steps inside the two docker files the dev version mimics running the code with `npm run dev` and doesn't really optimize the image for production. The production version will generate a highly optimized & lightweight container by running `npm run build` and then serving it with nginx.

Reply:
I understand the Dockerfile is meant to set up the project folder as an image. How is this image deployed to the hub? Do we have access to images deployed at ACM's hub. Are they protected from external users?

Reply:
> How is this image deployed to the hub?

We have a GitHub action that automatically deploys the image to DockerHub & GitHub packages. Check my reply to @Stevenmqnguyen's comment about secrets for more insights about that. 

> Do we have access to images deployed at ACM's hub.
Yes! Check out our [docker hub page](https://hub.docker.com/orgs/acmutd) & our [Github Packages](https://github.com/orgs/acmutd/packages). Both places serve the same purpose --> To host images in the cloud where anyone can access them. The GitHub Actions currently push to both because I'm still trying to figure out what the best pipeline is for our CI/CD process

> Are they protected from external users?

No. Anyone can access these images. Which you might think is a problem and that is correct. Anyone with a copy of the image will essentially be able to run it to get access to all the functionality of the website anywhere they want. WIth the two published images on DockerHub & GitHub Packages that is alright because we don't have any embdedded secrets in it (meaning besides the homepage nothing works). In the future once those secrets do get added in we might solely only decide to use the GCP Cloud Registry because images on there are private and can only be accessed by us. Both our current deployments on Cloud Run and Kubernetes already pull their image from GCP Cloud Registry.